src/Symfony/Component/Security/Http/Tests/Authentication/AuthenticatorManagerTest.php

```diff
@@ -328,11 +371,53 @@ public function testAuthenticateRequestHidesInvalidUserExceptions()
             ->with($this->equalTo($this->request), $this->callback(fn ($e) => $e instanceof BadCredentialsException && $invalidUserException === $e->getPrevious()))
             ->willReturn($this->response);
 
-        $manager = $this->createManager([$authenticator]);
+        $manager = $this->createManager([$authenticator], exposeSecurityErrors: true);
         $response = $manager->authenticateRequest($this->request);
         $this->assertSame($this->response, $response);
     }
 
+    public function testAuthenticateRequestShowsAccountStatusException()
+    {
+        $invalidUserException = new LockedException();
+        $authenticator = $this->createMock(TestInteractiveAuthenticator::class);
+        $this->request->attributes->set('_security_authenticators', [$authenticator]);
+
+        $authenticator->expects($this->any())->method('authenticate')->willThrowException($invalidUserException);
+
+        $authenticator->expects($this->any())
+            ->method('onAuthenticationFailure')
+            ->with($this->equalTo($this->request), $this->callback(fn ($e) => $e === $invalidUserException))
+            ->willReturn($this->response);
+
+        $manager = $this->createManager([$authenticator], exposeSecurityErrors: AuthenticatorManager::EXPOSE_ACCOUNT_STATUS);
+        $response = $manager->authenticateRequest($this->request);
+        $this->assertSame($this->response, $response);
+    }
+
+    /**
+     * @group legacy
+     */
+    public function testAuthenticateRequestHidesInvalidAccountStatusException()
+    {
+        $invalidUserException = new LockedException();
+        $authenticator = $this->createMock(TestInteractiveAuthenticator::class);
+        $this->request->attributes->set('_security_authenticators', [$authenticator]);
+
+        $authenticator->expects($this->any())->method('authenticate')->willThrowException($invalidUserException);
+
+        $authenticator->expects($this->any())
+            ->method('onAuthenticationFailure')
+            ->with($this->equalTo($this->request), $this->callback(fn ($e) => $e instanceof BadCredentialsException && $invalidUserException === $e->getPrevious()))
+            ->willReturn($this->response);
+
+        $manager = $this->createManager([$authenticator], exposeSecurityErrors: true);
+        $response = $manager->authenticateRequest($this->request);
+        $this->assertSame($this->response, $response);
+    }
+
+    /**
+     * @group legacy
```

Comment:
This should not be a legacy test, as we don't want to remove that test when removing the deprecated code.